docs/design/settings/extensions/settings-extension-example.md

```diff
@@ -0,0 +1,156 @@
+```
+├── packages
+│   └── mydaemon
+│       ├── build.rs
+│       ├── Cargo.toml
+│       ├── mydaemon.spec
+│       ├── mydaemon.template.yaml
+│       └── pkg.rs
+├── sources
+└── variants
+
+# mydaemon.template.toml
+---
+[required-extensions]
+mydaemon: v1
+---
+query-url: {{ mydaemon.query-url }}
+```
+The configuration template includes "frontmatter" which informs the settings system to use settings values owned by the `mydaemon` extension, then the rest of the template is used to render the configuration file.
+
+## Service File
+Changing settings typically results in changes to configuration file templates, which results in systemd services being restarted to trigger the change in behavior.
+You should specify a `mydaemon.service` systemd service file which will be installed into the Bottlerocket system and used to manage the lifecycle of your daemon.
+
+```
+├── packages
+│   └── mydaemon
+│       └── ...
+│       └── mydaemon.service
+├── sources
+└── variants
+```
+
+## Resulting Disk Layout
+We've discussed constructing a setting, using a configuration file template, and associating that with a service.
+The Bottlerocket build system must arrange these files on disk to produce an image file.
+This section discusses how the various artifacts will be arranged on a running system.
+
+### Settings Extension Binaries
+Extension binaries are stored under `sys-root/usr/libexec` and symlinked into `sys-root/usr/libexec/settings`.
+`sys-root/usr/share/settings` will also hold the config file for the settings extensions, which will be named `${SETTING_NAME}.toml`
+The setting owned by the extension will share a name with the symlink.
+
+
+For example, the extension for `my-setting` may be installed like so:
+```
+sys-root
+└── usr
+    ├── libexec
+    │   ├── my-setting-binary
+    │   └── settings
+    │       └── my-setting -> ../my-setting-binary
+    └── share
+        └── settings
+            └── my-setting.toml
+```
+
+### Configuration Templates
+
+Templates will continue to be stored in their location in current Bottlerocket under `sys-root/usr/share/templates`; however, the format will change to support arbitrarily associating these templates services.
+
+In the below case, we've configured the `motd.service` service to be restarted whenever the `motd` template is re-rendered.
+This is done because the service file has been symlinked into `templates/motd.template.affected-services/`.
+
+```
+sys-root
+└── usr
+    └── share
+        └── templates
+            ├── motd.template
+            ├── motd.template.affected-services
+            │       └── motd.service -> /etc/systemd/system/motd.service
+            └── motd.template.rendered-to
+                    └── motd.conf
+```
+
+`motd.template.rendered-to/motd.conf` contains instructions similar to `systemd-tmpfiles.d` explaining locations to which the rendered template will be placed.
+The format of this file dictates the resulting file location and mode of the resulting file:
```

Comment:
Clever, I missed the implications of this before. I like reusing part of the `tmpfiles.d` syntax for this. It might be good to also support setting UID / GID in addition to mode.
```
/etc/mydaemon/mydaemon.json - - -
/etc/mydaemon.env 0755 root root
```